src/libstore/file-ingestion-method.cc

```diff
@@ -0,0 +1,18 @@
+#include "file-ingestion-method.hh"
+#include "error.hh"
+
+namespace nix {
+
+std::string makeFileIngestionPrefix(FileIngestionMethod m)
+{
+    switch (m) {
+    case FileIngestionMethod::Flat:
+        return "";
+    case FileIngestionMethod::Recursive:
+        return "r:";
+    default:
+        throw Error("impossible, caught both cases");
+    }
+}
+
+}
```

Comment:
Thanks for not creating this in `libutils`, but could you just not create this file at all, and leave `content-address.{hh,cc}` the way it is? And likewise leave `file-content-address.{cc,hh}` in libutil the way they are? I don't think these changes are needed for the rest.